probe-rs/src/vendor/nxp/sequences/nxp_armv8m.rs

```diff
@@ -489,22 +501,43 @@ impl MIMXRT5xxS {
     }
 
     fn reset_flash(&self, interface: &mut dyn ArmProbe) -> Result<(), ArmError> {
-        tracing::trace!("MIMXRT595S-EVK FlexSPI flash reset (pulse PIO4_5)");
-
-        // FIXME: We do this by twiddling PIO4_5, which is where the flash
-        // reset pin is connected on MIMX595-EVK, but this code should not
-        // make any assumptions about the evaluation board; how can we
-        // generalize this so that the reset is configurable?
-        interface.write_word_32(0x40001044, 1 << 24)?; // enable GPIO clock
-        interface.write_word_32(0x40000074, 1 << 24)?; // take GPIO out of reset
-        interface.write_word_32(0x40004214, 0x130)?; // full drive and pullup
-        interface.write_word_32(0x40102010, 1 << 5)?; // PIO4_5 is an output
-        interface.write_word_32(0x40103214, 0)?; // PIO4_5 is driven low
-        std::thread::sleep(Duration::from_millis(100));
-
-        interface.write_word_32(0x40102010, 0)?; // PIO4_5 is an input
-        interface.flush()?;
-        std::thread::sleep(Duration::from_millis(100));
+        if self.family == MIMXRTFamily::MIMXRT5 {
+            tracing::trace!("MIMXRT595S-EVK FlexSPI flash reset (pulse PIO4_5)");
+
+            // FIXME: We do this by twiddling PIO4_5, which is where the flash
+            // reset pin is connected on MIMX595-EVK, but this code should not
+            // make any assumptions about the evaluation board; how can we
+            // generalize this so that the reset is configurable?
+            interface.write_word_32(0x40001044, 1 << 24)?; // enable GPIO clock
```

Comment:
I would prefer hiding the magic numbers behind named constants (i.e. `GPIO_BASE` and `GPIO_CLOCK_OFFSET`, `GPIO_CLK_ENABLE_BIT`, etc.). Those named constants can then be moved to the particular MCU family, and then it looks like this if-else may go away entirely.